apiserver/application/application.go

```diff
@@ -1049,18 +1049,14 @@ func (api *API) sameControllerRemoteApplicationInfo(url jujucrossmodel.Applicati
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	icon, err := api.charmIcon(NewStateBackend(st), ch.URL())
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
 	return &params.RemoteApplicationInfo{
 		ModelTag:         sourceModelTag.String(),
 		Name:             application.Name(),
 		Description:      ch.Meta().Description,
 		ApplicationURL:   url.String(),
 		SourceModelLabel: model.Name(),
 		Endpoints:        endpoints,
-		Icon:             icon,
+		IconURLPath:      fmt.Sprintf("rest/1.0/remote-application/%s/icon", url.ApplicationName),
```

Comment:
url.ApplicationName is not guaranteed to be the same as app.Name() but it is right now.
What the outside world sees is what's in the URL - that's the external interface so to speak.

With cross controller, 1. we don't support that yet (not for this cycle), 2. it is expected requests would still come in to the controller hosting the remote app reference as the gui is not guaranteed to have connectivity to the other controller. I could include the "model/:uuid" part of the URI in the path but as I understand it, the GUI uses hrefs relative to the base model URL for a given canvas, for all model facades, which Application is. In the JIMM case, request would come into the host controller and then be routed via JIMM proxy, so this fits with that.